packages/x-date-pickers/src/internals/hooks/useDesktopPicker/useDesktopPicker.types.ts

```diff
@@ -79,9 +80,13 @@ export interface ExportedUseDesktopPickerSlotProps<
     {},
     PickerOwnerState
   >;
-  textField?: SlotComponentProps<typeof TextField, {}, Record<string, any>>;
+  textField?: SlotComponentPropsFromProps<
```

Comment:
Always great to have this kind of cleaning :ok_hand: 
Now the `slots` exposed at the picker level will always have the right `ownerState` and the right props.